modules/core/src/main/java/org/apache/ignite/internal/util/CollectionUtils.java

```diff
@@ -197,13 +202,100 @@ public T next() {
         };
     }
 
+    /**
+     * Create a lazy concatenation of iterators.
+     *
+     * <p>NOTE: {@link Iterator#remove} - not supported.
+     *
+     * @param iterators Iterators.
+     * @param <T> Type of the elements.
+     * @return Concatenation of iterators.
+     */
+    @SafeVarargs
+    public static <T> Iterator<T> concat(@Nullable Iterator<? extends T>... iterators) {
```

Comment:
Why don't we use streams?

Reply:
I wrote above about the impossibility of using streams now.